Add chart plugin

Zerquix18 opened the pull request:
My submission for the [snapshot plugin bounty](https://gitcoin.co/issue/snapshot-labs/snapshot/577/100026411). There's a Youtube video to demo this plugin: https://youtu.be/Ax_68hQMs6U

I'm also submitting another pull request to the [other repository](https://github.com/snapshot-labs/snapshot.js/pull/361) with the plugin class.

Dependencies installed:
- chart.js

This submission delivers 3 charts to visualize the votes in a proposal:
1. Total votes per day - shows the total amount of votes added in a particular day
2. Share of voting power - how much voting power each address added, sorted in descending order in a pier chart.
3. Total voting power throughout time for each option

It also includes strings for English and Spanish, but for the other languages it uses English.

![image](https://user-images.githubusercontent.com/2019162/131774622-57c45b51-d6aa-4ad7-98ff-0ec3f869d66f.png)
![image](https://user-images.githubusercontent.com/2019162/131774639-9d211d58-dd6a-4656-89fd-27762d18d2a7.png)
![image](https://user-images.githubusercontent.com/2019162/131774651-2d1648d4-0d10-4388-8423-dbe7cda10409.png)

The demo video explains how each of these can be useful and helps you visualize them.

A few caveats:
1. This may not work right away since I hardcoded (but didn't commit) the plugin class to this repository to test. The version I'm submitting relies on my other pull request being accepted and deployed on npm.
2. In the thanku.eth space (which I used to test), the page may break since two people were working on a plugin that doesn't exist in the snapshot.js list but exists in the database (once saved) . I added a check while I was testing but didn't think it was necessary to commit that as well. [This is what I'm referring to](https://github.com/snapshot-labs/snapshot/blob/develop/src/views/Settings.vue#L88). 
3. I am not a Vue developer, I've only used it once besides now. Tried my best to keep it clean but my thinking process comes from React.

Hopefully people find this useful :) Thanks for the opportunity.

samuveth commented on src/locales/ar-SA.json:
```diff
@@ -273,5 +273,9 @@
+  },
+  "charts": {
+    "charts": "Charts",
+    "noVotesYet": "There are no votes to visualize yet."
```
Hi @Zerquix18, thanks for your contribution! You don't need to add this to every translation file, only default.json is enough.

Zerquix18 replied:
Removed them for all files except english and spanish. :)

samuveth replied:
Thanks. Though the Spanish translation should be added through translate.snapshot.org instead of here.

Zerquix18 replied:
Sure. I assume that has to be done if the pull request is merged, right?

samuveth replied:
No, but you can do it of you like.

samuveth commented on src/components/Plugin/Charts/CustomBlock.vue:
```diff
@@ -0,0 +1,189 @@
+
+const space = props.space;
+const proposal = props.proposal;
+const votes = props.votes;
```
This should be put inside a computed property.

You can do it like this: `const votes = computed(() => props.votes.sort((a, b) => a.created - b.created)))

Zerquix18 replied:
Pushed this update.

samuveth commented:
@Zerquix18 can you add me to your repo, I'd like to add some fixes.

Also I'm not seeing the charts in my proposal when testing locally. This is what I see:
![image](https://user-images.githubusercontent.com/51686767/131940789-1f8e29ca-993f-4333-b8cb-1a5a921958dc.png)

These are my settings:
![image](https://user-images.githubusercontent.com/51686767/131940839-4f0a2ea0-1dff-49dc-8825-02ac08af8630.png)

Zerquix18 commented:
@samuveth Added you to the repo.
Can you share the link of your test proposal so I can see the error?

samuveth commented:
> @samuveth Added you to the repo.
> Can you share the link of your test proposal so I can see the error?

Hey sorry I was busy with some other things. I will give this another try today

bonustrack commented:
Hi thanks for your contribution. Could you lazy load the package `charts.js`? This is something that will be used only on space that enabled the plugin it's better if this dont get loaded for all users but only when needed.

Zerquix18 commented:
Hi @bonustrack  thanks for the feedback. I added an update and now chart.js package will be loaded after the component is mounted.

With regards to @samuveth 's issue, I realized the settings have `enabled` while the code checked for `enable`. I think that's a typo in my part so now the code will check for `enabled`. I'm not sure if this is the best way to structure the settings. Let me know what you think.

I added the plugin to the thanku.eth space which I'm using to test, so it should be visible in the proposals there.
Locally, I'm adding this code:

```javascript
export default class Plugin {
  public author = 'zerquix18';
  public version = '0.1.0';
  public name = 'Charts';
}

// @ts-ignore
plugins.charts = Plugin;
```

to `useSearchFilters.ts` since the plugin is not on the other repo yet.
Let me know if you encounter any other issue :)

samuveth commented:
Hey @Zerquix18 I made a pull request on your branch with some changes.

samuveth commented:
Otherwise it works well for me! Great work!

Zerquix18 commented:
@samuveth  Tested on my computer and it works great. Thanks for reviewing!

samuveth commented:
@Zerquix18 can you make the charts more compact, like bonustrack mentioned already?

samuveth commented:
@Zerquix18 the top and bottom padding are too big on the 2nd chart, see below:
![image](https://user-images.githubusercontent.com/51686767/134483273-95889bbe-e80f-4d21-bf8e-eac500b68927.png)

Also can you make the chart title to be link-color?

Zerquix18 commented:
@samuveth @bonustrack 
Now it's separated into tabs & reduced the size of the 2nd chart.

samuveth commented:
@Zerquix18 can you resolve the conflicts?

Zerquix18 commented:
Should be good now

samuveth commented:
@Zerquix18 sorry happened again..

bonustrack commented:
Hey sorry last suggestions:
1: Title of the tab is not correct (duplicate of the first tab), could you fix it?
![image](https://user-images.githubusercontent.com/16245250/135723924-7cc5b137-425c-4c0f-94d1-9ddccf0c456f.png)
2: Could you remove the title in the charts, we can see the title in the tab already I dont think we need to repeat it, i see it overlap sometime like this
![image](https://user-images.githubusercontent.com/16245250/135724022-fa7d023f-ceb3-498d-9243-89eae2379b03.png)

bonustrack commented:
^ @Zerquix18

samuveth commented:
@Zerquix18 would be great if you could finish off this PR. If you don't have time please let us know
